### PR TITLE
[ESPnet2] Refactor Encoder and Decoder and bug fix

### PR DESCRIPTION
**Pull request by kamo-naoyuki**
- `--encder-decoder` -> `--encoder`, `--decoder`
- I split RNNEncoder into RNNEncoder and VGGRNNEncoder
- Renames some arguments. We should use more understandable names.
- I also found that vggrnn used subsampling in espnet2, but the ASR result in voxforge is still worse after fixing it. Anyone help me.

### CODE REVIEW

**Comment by kamo-naoyuki** [comment]
- ESPnet2

acc


```
2019-12-13T17:24:47 (asr.sh:545:main) Write cer result in exp/asr_train_fbank_pitch_char/decode_dt_it/score_cer/result.txt
| SPKR                  | # Snt  # Wrd | Corr    Sub    Del    Ins    Err  S.Err |
| Sum/Avg               | 1082   79133 | 88.5    6.2    5.3    2.6   14.2   98.5 |
2019-12-13T17:24:48 (asr.sh:545:main) Write wer result in exp/asr_train_fbank_pitch_char/decode_dt_it/score_wer/result.txt
| SPKR                  | # Snt # Wrd | Corr    Sub    Del    Ins    Err  S.Err |
| Sum/Avg               | 1082  13235 | 55.5   37.7    6.7    4.3   48.8   98.5 |

2019-12-13T17:24:49 (asr.sh:545:main) Write cer result in exp/asr_train_fbank_pitch_char/decode_et_it/score_cer/result.txt
| SPKR                  | # Snt  # Wrd | Corr    Sub    Del    Ins    Err  S.Err |
| Sum/Avg               | 1055   77966 | 88.5    6.3    5.2    2.5   14.0   98.7 |
2019-12-13T17:24:49 (asr.sh:545:main) Write wer result in exp/asr_train_fbank_pitch_char/decode_et_it/score_wer/result.txt
| SPKR                  | # Snt # Wrd | Corr    Sub    Del    Ins    Err  S.Err |

```

- ESPnet1

acc


```
write a CER (or TER) result in exp/tr_it_pytorch_train/decode_dt_it_decode/result.txt
| SPKR                  | # Snt  # Wrd | Corr    Sub    Del    Ins    Err  S.Err |
| Sum/Avg               | 1082   79133 | 89.7    5.7    4.6    2.5   12.8   97.9 |
write a WER result in exp/tr_it_pytorch_train/decode_dt_it_decode/result.wrd.txt
| SPKR                  | # Snt # Wrd | Corr    Sub    Del    Ins    Err  S.Err |
| Sum/Avg               | 1082  13235 | 58.2   36.2    5.5    5.0   46.8   97.9 |

write a CER (or TER) result in exp/tr_it_pytorch_train/decode_et_it_decode/result.txt
| SPKR                  | # Snt  # Wrd | Corr    Sub    Del    Ins    Err  S.Err |
| Sum/Avg               | 1055   77966 | 89.8    5.6    4.5    2.5   12.7   98.0 |
write a WER result in exp/tr_it_pytorch_train/decode_et_it_decode/result.wrd.txt
| SPKR                  | # Snt # Wrd | Corr    Sub    Del    Ins    Err  S.Err |
| Sum/Avg               | 1055  12990 | 59.1   35.2    5.7    5.8   46.7   98.0 |
```

**Comment by kamo-naoyuki** [comment]
I found `loss.best` model is used for decoding by default, and I changed it to `acc.best`.

```
2019-12-13T17:46:39 (asr.sh:545:main) Write cer result in exp/asr_train_fbank_pitch_char/decode_dt_it/score_cer/result.txt
| SPKR                  | # Snt  # Wrd | Corr    Sub    Del    Ins    Err  S.Err |
| Sum/Avg               | 1082   79133 | 89.4    5.9    4.8    2.9   13.5   98.6 |
2019-12-13T17:46:39 (asr.sh:545:main) Write wer result in exp/asr_train_fbank_pitch_char/decode_dt_it/score_wer/result.txt
| SPKR                  | # Snt # Wrd | Corr    Sub    Del    Ins    Err  S.Err |
| Sum/Avg               | 1082  13235 | 57.4   37.3    5.3    5.5   48.1   98.6 |
2019-12-13T17:46:40 (asr.sh:545:main) Write cer result in exp/asr_train_fbank_pitch_char/decode_et_it/score_cer/result.txt
| SPKR                  | # Snt  # Wrd | Corr    Sub    Del    Ins    Err  S.Err |
| Sum/Avg               | 1055   77966 | 89.7    5.8    4.5    2.7   13.1   98.7 |
2019-12-13T17:46:41 (asr.sh:545:main) Write wer result in exp/asr_train_fbank_pitch_char/decode_et_it/score_wer/result.txt
| SPKR                  | # Snt # Wrd | Corr    Sub    Del    Ins    Err  S.Err |
| Sum/Avg               | 1055  12990 | 57.7   37.0    5.4    5.7   48.0   98.7 |
```

File: egs2/voxforge/asr1/run.sh [hunk 1]
```diff
@@ -10,11 +10,18 @@ train_set="tr_${lang}"
...
+asr_config=conf/asr_rnn/train.yaml
+decode_config=conf/asr_rnn/decode.yaml
+
 
 ./asr.sh \
     --local_data_opts "--lang ${lang}" \
     --token_type char \
-    --lm_config conf/lm_train.yaml \
+    --feats_type fbank_pitch \
+    --lm_config "${lm_config}" \
```
Comment (sw005320) [inline egs2/voxforge/asr1/run.sh]: why did you add lm and decode configs?

Comment (kamo-naoyuki) [reply egs2/voxforge/asr1/run.sh]: just for example

l  ll add an option to skip lm training and decode without lm.
